media_service/models.py

```diff
@@ -164,9 +168,11 @@ class Meta:
     def save(self, *args, **kwargs):
         if not self.sort_order:
             self.sort_order = self.next_sort_order({"course__pk": self.course.pk})
+        if not (isinstance(self.metadata, basestring) and len(self.metadata) > 0 and '[]' == self.metadata[0] + self.metadata[-1]):
```

Comment:
Although somewhat more resource-intensive, an alternative (which might be easier to read) would be to do something like (maybe wrap in a helper method):
```
try:
     self.metadata = self.metadata if isinstance(json.loads(self.metadata), list) else metadata_default()
except (TypeError, ValueError):  # or you could just catch everything
     self.metadata = metadata_default()
```
If you expect/want to support malformed JSON, that's a different story.
You could also use Django's model implementation of the Postgresql JSON field.... :)